uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt311/client/paho/Mqtt311ConnectionImpl.java

```diff
@@ -222,26 +226,9 @@ private static ConnectResult buildConnectResult(boolean success, Boolean session
 
     private class MqttMessageListener implements IMqttMessageListener {
 
-        @SuppressWarnings("PMD.AvoidCatchingGenericException")
         @Override
-        public void messageArrived(String topic, MqttMessage mqttMessage) throws Exception {
-            if (isClosing.get()) {
-                logger.atWarn().log("PIBLISH event ignored due to shutdown initiated");
-            } else {
-                GRPCClient.MqttReceivedMessage message = new GRPCClient.MqttReceivedMessage(
-                        mqttMessage.getQos(), mqttMessage.isRetained(), topic, mqttMessage.getPayload(),
-                        null, null, null,null, null, null);
-                executorService.submit(() -> {
-                    try {
-                        grpcClient.onReceiveMqttMessage(connectionId, message);
-                    } catch (Exception ex) {
-                        logger.atError().withThrowable(ex).log("onReceiveMqttMessage failed");
-                    }
-                });
-
-                logger.atInfo().log("Received MQTT message: connectionId {} topic '{}' QoS {} retain {}",
-                                        connectionId, topic, mqttMessage.getQos(), mqttMessage.isRetained());
-            }
+        public void messageArrived(String topic, MqttMessage mqttMessage) {
+            Mqtt311ConnectionImpl.this.messageArrived(topic, mqttMessage);
```

Comment:
1. Does wrong method will be called without `Mqtt311ConnectionImpl.this`  ? 

2. Looks like infinite recursion.

Reply:
fixed in dfd2cb402305dcd47a03070e302a2a1f0592d8f7